sound/soc/intel/boards/Kconfig

```diff
@@ -261,6 +261,7 @@ config SND_SOC_INTEL_DA7219_MAX98357A_GENERIC
 	select SND_SOC_DA7219
 	select SND_SOC_MAX98357A
 	select SND_SOC_DMIC
+	select SND_HDA_CODEC_HDMI if SND_SOC_SOF_HDA_CODEC
```

Comment:
this looks odd

config SND_SOC_SOF_HDA_COMMON_HDMI_CODEC
	bool "SOF common HDA HDMI codec driver"
	depends on SND_SOC_SOF_HDA_LINK
	depends on SND_HDA_CODEC_HDMI
	default y if SND_HDA_CODEC_HDMI

and here we test for HDA_CODEC?

Reply:
@plbossart I tried multiple things, but I kept getting problems with the default/generated configs (resulting in build errors due to missing symbols). Specifically neither SND_HDA_CODEC_HDMI nor SND_SOF_SOF_HDA_LINK guarantee that the core SND_HDA is enabled. By depending on SND_SOC_SOF_HDA_CODEC in the machine drivers, I can guarantee that all dependencies are enabled for CODEC_HDMI. Some of the machine drivers are shared with SST and there common codec is never used, so thus the SOF-specific dependency.

It's a bit too complicated I agree, but if we want to keep the "enabling machine driver enables all dependencies" feature, we need to something like this.

Reply:
We shouldn't tie CODEC_HDMI and SOF_HDA_CODEC. we want to be able to use the common HDMI stuff even when we don't have on-board HDAudio codecs.

Reply:
@plbossart Hmm, that doesn't really work, HDMI is now just another HDA codec and we use same codepaths. E.g. in hda-codec.c, hda_codec_probe() will fail for HDMI unless SOF_HDA_CODEC is set.

Reply:
@kv2019i if we want to only enable the HDMI codec, and NOT the HDAudio one, do we have a solution with the new stuff?

one example will be the Up Extreme, which has an HDaudio codec on board. I don't necessarily want to use it, so can we ignore it with Kconfigs?

Reply:
@plbossart Hmm, I believe yes. You can disable the HDaudio codec in question, or just blacklist it via modprobe.conf (e.g. snd-hda-codec-realtek). The machine driver and topology need to match (and not include DAI links for the blacklisted HDA codec). Tested on CFL with a modified machine driver and sof-hda-generic-idisp.tplg

Reply:
what I meant is disable all HDAudio codecs except for iDISP as a first pass. then when HDA_CODEC is set we all all non-iDISP codecs.

Reply:
@plbossart I get the ask, but that's not possible now and would need a new Kconfig option. So what you can do now is to blacklist (or disable in Kconfig) all non-idisp codecs and get what you want, but we are definitely lacking capability to do this with a SOF Kconfig (HDA_CODEC_IDISP_ONLY=y or some such). Should we file a enhancement for this? I don't think this is directly coupled with the patch series (common hdmi codec is already merged, this is just changing the defaults).

Reply:
@kv2019i For debug/developement, it's a really useful feature to take out external on-board HDAudio codecs. at some point I also wanted to remove iDISP (e.g. to work around some of the bad masks that can be reported at times), while keeping the HDaudio codec accessible.

If we can't do it with existing Kconfig, let's try and improve this with a follow-up PR. Thanks.

Reply:
@plbossart Ok, tried a few options and one promising option is to add a "force_codec_mask" module parameter. The legacy driver has this and based on a quick hack, seems to work nicely on SOF as well. I.e. I can force codec_mask to ignore the external HDA codec, and SOF goes to ignore and sets up the HDMI PCM paths only. Sounds good? I can make a separate PR about this.

And how about the rest of the series, seems ok? I added a few more people to review list.

Reply:
sounds good. we could have an explicit kconfig that sets the mask directly as a default value, and let the user override it.

A different PR is probably better, thanks!